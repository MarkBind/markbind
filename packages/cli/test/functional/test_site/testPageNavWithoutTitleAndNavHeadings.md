<frontmatter>
  layout: default.md
  pageNav: 1
</frontmatter>

There should not be any contents in the page nav component/section of the page.

Hence, the separator i.e., `border-left: ...` should not be present in the above mentioned component.
