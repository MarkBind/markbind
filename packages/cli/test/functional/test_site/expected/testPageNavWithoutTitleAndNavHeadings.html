<!DOCTYPE html>
<html>

<head>

  <meta name="default-head-top">
  <script src="/test_site/headFiles/customScriptTop.js"></script>

  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="MarkBind 5.4.0">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>page nav separator should be omitted when there is no title and navigation headings added</title>
  <link rel="stylesheet" href="/test_site/markbind/css/bootstrap.min.css">
  <link rel="stylesheet" href="/test_site/markbind/fontawesome/css/all.min.css">
  <link rel="stylesheet" href="/test_site/markbind/glyphicons/css/bootstrap-glyphicons.min.css">
  <link rel="stylesheet" href="/test_site/markbind/css/codeblock-dark.min.css">
  <link rel="stylesheet" href="/test_site/markbind/css/markbind.min.css">
  <link rel="stylesheet" href="/test_site/plugins/testMarkbindPlugin/testMarkbindPluginStylesheet.css">
  <link rel="stylesheet" href="/test_site/plugins/web3Form/web-3-form.css">
  <link rel="stylesheet" href="/test_site/plugins/markbind-plugin-anchors/markbind-plugin-anchors.css">
  <link rel="stylesheet" href="/test_site/plugins/markbind-plugin-tree/markbind-plugin-tree.css">

  <meta name="default-head-bottom">
  <link rel="stylesheet" href="/test_site/stylesheets/styles.css">
  <script src="/test_site/headFiles/customScriptBottom.js"></script>
  <link rel="icon" href="/test_site/favicon.png">
</head>
<script>
  const baseUrl = '/test_site'
</script>

<body>
  <div id="app">

    <div>
      <header>
        <navbar type="dark" default-highlight-on="sibling-or-child">
          <template #brand><a href="/" title="Home" class="navbar-brand">MarkBind Test Site</a></template>
          <li><a class="nav-link" href="/test_site/bugs/index.html">Open Bugs</a></li>
        </navbar>
        <div class="bg-info display-4 text-center text-white">
          <br>
          Test Jumbotron<br>
          <br>
        </div>
      </header>
      <p><strong>Relative Link Test</strong> This is a relative Intra-Site link in a layout (see <a href="/test_site/index.html#heading-with-hidden-keyword">link</a>)</p>
    </div>
    <div id="flex-body">
      <overlay-source id="site-nav" tag-name="nav" to="site-nav">
        <div class="site-nav-top">
          <div class="fw-bold mb-2" style="font-size: 1.25rem;">
            <div>
              <h2 id="default-layout">Default Layout<a class="fa fa-anchor" href="#default-layout" onclick="event.stopPropagation()"></a></h2>
            </div>
          </div>
        </div>
        <div class="nav-component slim-scroll">
          <div>
            <site-nav>
              <overlay-source class="site-nav-list site-nav-list-root" tag-name="ul" to="mb-site-nav">
                <li>
                  <div class="site-nav-default-list-item site-nav-list-item-0" onclick="handleSiteNavClick(this)"><a href="/test_site/index.html">Home üè†</a></div>
                </li>
                <li>
                  <div class="site-nav-default-list-item site-nav-list-item-0" onclick="handleSiteNavClick(this)"><a href="/test_site/bugs/index.html">Open Bugs üêõ</a></div>
                </li>
                <li class="site-nav-custom-list-item site-nav-list-item-0">
                  <h3 id="testing-site-nav">Testing Site-Nav<a class="fa fa-anchor" href="#testing-site-nav" onclick="event.stopPropagation()"></a></h3>
                </li>
                <li>
                  <div class="site-nav-default-list-item site-nav-list-item-0" onclick="handleSiteNavClick(this)"><strong>Dropdown </strong> <span aria-hidden="true" class="glyphicon glyphicon-search"></span> title ‚úèÔ∏è
                    <!-- Nested list items will be placed inside a Dropdown menu -->

                    <div class="site-nav-dropdown-btn-container"><i class="site-nav-dropdown-btn-icon site-nav-rotate-icon" onclick="handleSiteNavClick(this.parentNode.parentNode, false); event.stopPropagation();">
                        <span class="glyphicon glyphicon-menu-down" aria-hidden="true"></span>
                      </i></div>
                  </div>
                  <ul class="site-nav-dropdown-container site-nav-dropdown-container-open site-nav-list">
                    <li>
                      <div class="site-nav-default-list-item site-nav-list-item-1" onclick="handleSiteNavClick(this)"><a href="https://www.google.com/">Dropdown link one</a></div>
                    </li>
                    <li>
                      <div class="site-nav-default-list-item site-nav-list-item-1" onclick="handleSiteNavClick(this)"><a href="https://www.google.com/">Html within site-nav <span style="color: red;">should</span> be displayed properly</a></div>
                    </li>
                    <li>
                      <div class="site-nav-default-list-item site-nav-list-item-1" onclick="handleSiteNavClick(this)">Nested Dropdown title üìê

                        <div class="site-nav-dropdown-btn-container"><i class="site-nav-dropdown-btn-icon" onclick="handleSiteNavClick(this.parentNode.parentNode, false); event.stopPropagation();">
                            <span class="glyphicon glyphicon-menu-down" aria-hidden="true"></span>
                          </i></div>
                      </div>
                      <ul class="site-nav-dropdown-container site-nav-list">
                        <li>
                          <div class="site-nav-default-list-item site-nav-list-item-2" onclick="handleSiteNavClick(this)"><a href="https://www.google.com/"><strong>Nested</strong> Dropdown link one</a></div>
                        </li>
                        <li>
                          <div class="site-nav-default-list-item site-nav-list-item-2" onclick="handleSiteNavClick(this)"><a href="https://www.google.com/"><strong>Nested</strong> Dropdown link two</a></div>
                        </li>
                      </ul>
                    </li>
                    <li>
                      <div class="site-nav-default-list-item site-nav-list-item-1" onclick="handleSiteNavClick(this)"><a href="https://www.google.com/">Dropdown link two</a></div>
                    </li>
                  </ul>
                </li>
                <li>
                  <div class="site-nav-default-list-item site-nav-list-item-0" onclick="handleSiteNavClick(this)"><a href="https://www.google.com/"><mark>Third Link</mark> üìã</a></div>
                </li>
                <li>
                  <div class="site-nav-default-list-item site-nav-list-item-0" onclick="handleSiteNavClick(this)">Filler text <a href="https://www.youtube.com/"><span aria-hidden="true" class="glyphicon glyphicon-facetime-video"></span> Youtube üì∫</a> filler text
                    <!-- Using a link as a Dropdown title will not render a Dropdown menu -->

                    <div class="site-nav-dropdown-btn-container"><i class="site-nav-dropdown-btn-icon" onclick="handleSiteNavClick(this.parentNode.parentNode, false); event.stopPropagation();">
                        <span class="glyphicon glyphicon-menu-down" aria-hidden="true"></span>
                      </i></div>
                  </div>
                  <ul class="site-nav-dropdown-container site-nav-list">
                    <li>
                      <div class="site-nav-default-list-item site-nav-list-item-1" onclick="handleSiteNavClick(this)"><a href="https://www.youtube.com/watch?v=dQw4w9WgXcQ">The answer to everything in the universe</a></div>
                    </li>
                    <li>
                      <div class="site-nav-default-list-item site-nav-list-item-1" onclick="handleSiteNavClick(this)"><mark>Dropdown title</mark> <span aria-hidden="true" class="glyphicon glyphicon-comment"></span> ‚úèÔ∏è
                        <!-- Dropdown menus in are still supported inside, as long as the title is not a link -->

                        <div class="site-nav-dropdown-btn-container"><i class="site-nav-dropdown-btn-icon site-nav-rotate-icon" onclick="handleSiteNavClick(this.parentNode.parentNode, false); event.stopPropagation();">
                            <span class="glyphicon glyphicon-menu-down" aria-hidden="true"></span>
                          </i></div>
                      </div>
                      <ul class="site-nav-dropdown-container site-nav-dropdown-container-open site-nav-list">
                        <li>
                          <div class="site-nav-default-list-item site-nav-list-item-2" onclick="handleSiteNavClick(this)"><a href="https://www.google.com/"><strong>Nested</strong> Dropdown link one</a></div>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li>
                  <div class="site-nav-default-list-item site-nav-list-item-0" onclick="handleSiteNavClick(this)">Really Long Dropdown Title Really Long Dropdown Title Really Long Dropdown Title Really Long Dropdown

                    <div class="site-nav-dropdown-btn-container"><i class="site-nav-dropdown-btn-icon" onclick="handleSiteNavClick(this.parentNode.parentNode, false); event.stopPropagation();">
                        <span class="glyphicon glyphicon-menu-down" aria-hidden="true"></span>
                      </i></div>
                  </div>
                  <ul class="site-nav-dropdown-container site-nav-list">
                    <li class="site-nav-custom-list-item site-nav-list-item-1">Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text Really Really Long Text</li>
                    <li>
                      <div class="site-nav-default-list-item site-nav-list-item-1" onclick="handleSiteNavClick(this)">Nested Dropdown Title

                        <div class="site-nav-dropdown-btn-container"><i class="site-nav-dropdown-btn-icon" onclick="handleSiteNavClick(this.parentNode.parentNode, false); event.stopPropagation();">
                            <span class="glyphicon glyphicon-menu-down" aria-hidden="true"></span>
                          </i></div>
                      </div>
                      <ul class="site-nav-dropdown-container site-nav-list">
                        <li class="site-nav-custom-list-item site-nav-list-item-2">Hello Doge Hello Doge üê∂</li>
                        <li>
                          <div class="site-nav-default-list-item site-nav-list-item-2" onclick="handleSiteNavClick(this)"><a href="/test_site/index.html"><strong>NESTED LINK</strong> Home üè†</a></div>
                        </li>
                        <li class="site-nav-custom-list-item site-nav-list-item-2">Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit Text cut off from height limit</li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li>
                  <div class="site-nav-default-list-item site-nav-list-item-0" onclick="handleSiteNavClick(this)">Test line break in navigation layout

                    <div class="site-nav-dropdown-btn-container"><i class="site-nav-dropdown-btn-icon" onclick="handleSiteNavClick(this.parentNode.parentNode, false); event.stopPropagation();">
                        <span class="glyphicon glyphicon-menu-down" aria-hidden="true"></span>
                      </i></div>
                  </div>
                  <ul class="site-nav-dropdown-container site-nav-list">
                    <li class="site-nav-custom-list-item site-nav-list-item-1">Nested line break text ‚úÇÔ∏è</li>
                    <li>
                      <div class="site-nav-default-list-item site-nav-list-item-1" onclick="handleSiteNavClick(this)"><a href="/test_site/index.html">Nested line break href</a>

                        <div class="site-nav-dropdown-btn-container"><i class="site-nav-dropdown-btn-icon" onclick="handleSiteNavClick(this.parentNode.parentNode, false); event.stopPropagation();">
                            <span class="glyphicon glyphicon-menu-down" aria-hidden="true"></span>
                          </i></div>
                      </div>
                      <ul class="site-nav-dropdown-container site-nav-list">
                        <li class="site-nav-custom-list-item site-nav-list-item-2">Nested Nested line break text ‚úÇÔ∏è</li>
                      </ul>
                    </li>
                    <li>
                      <div class="site-nav-default-list-item site-nav-list-item-1" onclick="handleSiteNavClick(this)">Nested line break dropdown menu

                        <div class="site-nav-dropdown-btn-container"><i class="site-nav-dropdown-btn-icon" onclick="handleSiteNavClick(this.parentNode.parentNode, false); event.stopPropagation();">
                            <span class="glyphicon glyphicon-menu-down" aria-hidden="true"></span>
                          </i></div>
                      </div>
                      <ul class="site-nav-dropdown-container site-nav-list">
                        <li class="site-nav-custom-list-item site-nav-list-item-2">Line break item 2 üìò</li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </overlay-source>
            </site-nav>
          </div>
        </div>
      </overlay-source>
      <div id="content-wrapper">
        <breadcrumb></breadcrumb>

        <p>There should not be any contents in the page nav component/section of the page.</p>
        <p>Hence, the separator i.e., <code class="line-numbers hljs inline no-lang" v-pre>border-left: ...</code> should not be present in the above mentioned component.</p>
      </div>
      <overlay-source id="page-nav" tag-name="nav" to="page-nav">
        <div class="nav-component slim-scroll">
        </div>
      </overlay-source>
      <scroll-top-button></scroll-top-button>
    </div>
    <div>
      <footer>
        <h1 id="heading-in-footer-should-not-be-indexed">Heading in footer should not be indexed<a class="fa fa-anchor" href="#heading-in-footer-should-not-be-indexed" onclick="event.stopPropagation()"></a></h1>
        <div class="text-center">
          This is a dynamic height footer that supports markdown <span>üòÑ</span>!
        </div>
      </footer>
    </div>
  </div>
</body>
<script src="/test_site/markbind/js/bootstrap-utility.min.js"></script>
<script src="/test_site/markbind/js/polyfill.min.js"></script>
<script src="/test_site/markbind/js/vue.min.js"></script>
<script src="/test_site/markbind/js/markbind.min.js"></script>
<script src="testPageNavWithoutTitleAndNavHeadings.page-vue-render.js"></script>
<script>
  MarkBind.setupWithSearch()
</script>
<script src="/test_site/plugins/testMarkbindPlugin/testMarkbindPluginScript.js"></script>
<script>alert("Inline plugin script loaded!")</script>

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=TRACKING-ID"></script>
<script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());
    
      gtag('config', 'TRACKING-ID');
    </script>

<script>
        function submitForm(element) {
            event.preventDefault();
            const formData = new FormData(element);
            const formProps = Object.fromEntries(formData);
            const json = JSON.stringify(formProps);
            const submitButton = element.querySelector('button[type="submit"]');
            const submitButtonText = submitButton.innerText;
            submitButton.innerText = 'Please wait...'
            fetch('https://api.web3forms.com/submit', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                    'Accept': 'application/json'
                },
                body: json,
            })
            .then(async (response) => {
                if (response.status == 200) {
                    alert('Form submitted! Thank you for your response');
                } else {
                    alert('Error submitting form! Please try again later.');
                }
            })
            .catch(error => {
                alert('Error submitting form! Please try again later.');
            })
            .finally(() => {
                submitButton.innerText = submitButtonText;
            })
        }
    </script>
<script type="module">
    import mermaid from 'https://unpkg.com/mermaid@10/dist/mermaid.esm.min.mjs';
    document.addEventListener('DOMContentLoaded', () => {
      mermaid.initialize({});
      Vue.directive('mermaid', {
        inserted: function(el) {
          mermaid.run({
            nodes: [el]
          });
        }
      });
    });
  </script>

</html>